crates/zksync/core/src/vm/factory_deps.rs

```diff
@@ -0,0 +1,80 @@
+use itertools::Itertools;
```

Comment:
This module and its associated functions should be part for `inspect.rs` as there is never a case where we wish to split transactions without running through `inspect_multi`. We could rename it to `inspect_as_batch`, keep the interface similar to `inspect`, and call `split_tx_by_factory_deps` internally to preserve cohesion.

Reply:
We can revisit this when we add that functionality. Right now, it's hard to justify separating these from their point of use. It might be a while before we add forge create support so it would make sense to follow the normal evolution of the code.